### ISSUE TITLE
user_model_fqn used in Register, but not in Login

### ISSUE THREAD

**Issue by Cadienvan** [Laravel-Backpack/Base#122]
Hi everyone,
I'm facing a serious problem with using a custom user model in Backpack.

I create a new model, migrated it based on basic laravel user model, and specified in config/backpack/base.php the new model name.

I found out this configuration is used in registration (A user is correctly created and I am correctly auto-logged after creation), but if I logout and try to login again it just says matching credentials were not found.

Tried many solutions(Creating guards, providers, controllers, even editing Backpack core), but I'm just getting nowhere.

Is there any help you could provide me? I need no guards, middlewares or whatever, I just need to use a different user model.

**Comment by jorenvanhee** [comment]
I'll try to figure this out today

**Comment by lloy0076** [comment]
Sounds like a Laravel problem, not backpack.

**Comment by jorenvanhee** [comment]
I made it work locally :)
Working on a PR :)

**Comment by jorenvanhee** [comment]
Ok, [this](https://github.com/jorenvanhee/Base/commit/9efabe3ebf0dbc2eab2ebf1c5fb03a643eb3f035) is what I have so far. I basically made it possible to switch guards. I still have to test everything, but most stuff works for now.

I'm kinda stuck in the views. I found this for instance in the views: `Auth::check()`. And that should become something like this: `Auth::guard('guard-from-config')->check()`. Ideally, I would just have a variable in the views `$backpackAuthenticated`. But I don't know how we can use something like view composers in a package.

Thoughts?

**Comment by lloy0076** [comment]
It's not clear to me why `Auth::check()` would need to become somewhat more complex or need to be changed.

**Comment by Cadienvan** [comment]
I'm just wondering why there's an option called "user_model_fqn" that allows you to properly define a registration model, but that get ignored when logging in. It just makes this option useless and misleading..

**Comment by lloy0076** [comment]
@Cadienvan - you have a point but if you're using the _default_ authentication drivers you need to configure the right model in `config/auth.php`.

**Comment by jorenvanhee** [comment]
So the `user_model_fqn` option is only used in backpack's `RegisterController`. If you want to change the model for logging in as well, you have to change the `model` option of the provider of laravel's default guard. The default guard is `web`, the provider of the `web` guard is `users`.

```php
/* config/auth.php */
'providers' => [
    'users' => [
        'driver' => 'eloquent',
        'model' => App\User::class,
    ],
```

This means that the `user_model_fqn` option is useless. The model could just be fetched from the providers config.

The problem starts when you want to use another guard that is not the default one. For instance, you have a guard for the users on the public facing part of the website. And you have a different guard for the admins of the website. So a `user` guard and an `admin` guard. If the `users` guard is the default one, then it is impossible to use the `admin` guard with backpack.

The `LoginController` uses the `AuthenticatesUsers` trait. This trait contains the following method.

```php
/**
 * Get the guard to be used during authentication.
 *
 * @return \Illuminate\Contracts\Auth\StatefulGuard
 */
protected function guard()
{
    return Auth::guard();
}
```

As you can see, the default guard is being used here. That's why I've overwritten this method in [my fork](https://github.com/jorenvanhee/Base/commit/9efabe3ebf0dbc2eab2ebf1c5fb03a643eb3f035):

```php
protected function guard()
{
    $guard = config('backpack.base.guard')
        ?: config('auth.defaults.guard');

    return Auth::guard($guard);
}
```

**Comment by lloy0076** [comment]
What you're saying makes sense - maybe suggest it as an actual PR?

**Comment by soham2008xyz** [comment]
I'm facing this exact same problem and stumbled upon this issue while looking for a solution. I have two separate guards for the user-facing and admin-facing parts of a project I am building, and I tried setting the `user_model_fqn` option in the configuration. However, after much hair-tearing and reading through the LoginController and associated views, I realised the default guard is being used in all cases, same as @jorenvanhee.

I'd love to help out with the PR to close this issue.

**Comment by jorenvanhee** [comment]
I've created a PR @soham2008xyz. Would be nice if you could check if it also works for you :)

**Comment by soham2008xyz** [comment]
Pulled your code, re-published config and views and set required options - works like a charm! @jorenvanhee

**Comment by tabacitu** [comment]
Completely agree with eliminating ```user_model_fqn``` if we can use the value from another config file. Would be a breaking change, though...

**Comment by Cadienvan** [comment]
Solution proposed by @jorenvanhee seems the best one, as far as I understand.
It would be probably easier to just stick with the default User Model both in Front-End and Backpack integration, and maybe put a switch for admin users identification (I did this in my current project, and rewritten auth controllers to check for it).

**Comment by Donkfather** [comment]
Hi. I think this one is related.. So .. I installed backpack created a user from the registration view and logged in. After this I changed the User Model to CmsUser and changed the fqd in base.php. I can still login with the account from the users table but the new one created was in the cms_users.

**Comment by lloy0076** [comment]
If this is **not** fixed by #165 or needs to be open, please reopen.
